src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -962,6 +984,19 @@ private void GeneratePropertyGetter<T>(QualifiedType returnType, T decl, Class @
                     ReturnType = decl.QualifiedType
                 };
 
+                var arrayType = field.Type as ArrayType;
+
+                if (arrayType != null && @class.IsValueType)
+                {
+                    CSharpTypePrinter typePrinter = new CSharpTypePrinter(ctx.Driver);
+                    string type = arrayType.CSharpType(typePrinter).Type;
+                    WriteLine(string.Format("fixed ({0} __arrPtr = {1}.{2})",
```

Comment:
The native context was a guess of mine which turned out to be incorrect. I am sorry for wasting your time. Please note, however, that you need to avoid manipulating strings unless there's no other way, and that's why asked you to try the printer.